cli/command/task_test.go

```diff
@@ -0,0 +1,295 @@
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestReadTask(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	taskCommand := NewTaskCommand(base.Command())
+	result := []*models.TaskSummary{
+		{TaskID: "task_id"},
+	}
+
+	base.Resolver.EXPECT().
+		Resolve("task", "task_name").
+		Return(Args{"task_id"}, nil)
+
+	base.Client.EXPECT().
+		ListTasks().
+		Return(result, nil)
+
+	base.Client.EXPECT().
+		ReadTask("task_id").
+		Return(&models.Task{}, nil)
+
+	c := NewContext(t, Args{"task_name"}, nil)
+	if err := taskCommand.read(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestReadTask_userInputErrors(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	taskCommand := NewTaskCommand(base.Command())
+
+	base.Client.EXPECT().
+		ListTasks().
+		Return([]*models.TaskSummary{}, nil)
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": NewContext(t, nil, nil),
+	}
+
+	for name, c := range contexts {
+		if err := taskCommand.read(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestReadTask_expiredTasks(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	taskCommand := NewTaskCommand(base.Command())
+	result := []*models.TaskSummary{
+		{TaskID: "task_id"},
+	}
+
+	base.Resolver.EXPECT().
+		Resolve("task", "task_name").
+		Return(Args{"task_id", "task_id2", "task_id3"}, nil)
+
+	base.Client.EXPECT().
+		ListTasks().
+		Return(result, nil)
+
+	base.Client.EXPECT().
+		ReadTask("task_id").
+		Return(&models.Task{}, nil)
+
+	c := NewContext(t, Args{"task_name"}, nil)
+	if err := taskCommand.read(c); err != nil {
+		t.Fatal(err)
+	}
+
+}
+
+func TestListTasks(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	taskCommand := NewTaskCommand(base.Command())
+
+	base.Client.EXPECT().
+		ListTasks().
+		Return([]*models.TaskSummary{}, nil)
+
+	c := NewContext(t, nil, nil)
+	if err := taskCommand.list(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestReadTaskLogs(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	taskCommand := NewTaskCommand(base.Command())
+
+	base.Resolver.EXPECT().
+		Resolve("task", "name").
+		Return(Args{"id"}, nil)
+
+	query := buildLogQueryHelper("id", "start", "end", 100)
+
+	base.Client.EXPECT().
+		ReadTaskLogs("id", query)
+
+	flags := Flags{
+		"tail":  100,
+		"start": "start",
+		"end":   "end",
+	}
+
+	c := NewContext(t, Args{"name"}, flags)
+	if err := taskCommand.logs(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestReadTaskLogs_userInputErrors(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	taskCommand := NewTaskCommand(base.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": NewContext(t, nil, nil),
+	}
+
+	for name, c := range contexts {
+		if err := taskCommand.logs(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestParseOverrides(t *testing.T) {
+	input := Args{
```

Comment:
using the `Args` type doesn't make sense here. (I'm guessing this was a `sed` command gone wrong?)

Reply:
Yup. My bad.